nilearn/decomposition/dict_learning.py

```diff
@@ -0,0 +1,201 @@
+
+    Parameters
+    ----------
+    mask: Niimg-like object or MultiNiftiMasker instance, optional
+        Mask to be used on data. If an instance of masker is passed,
+        then its mask will be used. If no mask is given,
+        it will be computed automatically by a MultiNiftiMasker with default
+        parameters.
+
+    data: array-like, shape = [[n_samples, n_features], ...]
+        Training vector, where n_samples is the number of samples,
+        n_features is the number of features. There is one vector per
+        subject.
+
+    n_components: int
+        Number of components to extract
+
+    smoothing_fwhm: float, optional
+        If smoothing_fwhm is not None, it gives the size in millimeters of the
+        spatial smoothing to apply to the signal.
+
+    standardize : boolean, optional
+        If standardize is True, the time-series are centered and normed:
+        their variance is put to 1 in the time dimension.
+
+    alpha: float, optional, default=1
+        Sparsity controlling parameter
+
+    random_state: int or RandomState
+        Pseudo number generator state used for random sampling.
+
+    target_affine: 3x3 or 4x4 matrix, optional
+        This parameter is passed to image.resample_img. Please see the
+        related documentation for details.
+
+    target_shape: 3-tuple of integers, optional
+        This parameter is passed to image.resample_img. Please see the
+        related documentation for details.
+
+    low_pass: None or float, optional
+        This parameter is passed to signal.clean. Please see the related
+        documentation for details
+
+    high_pass: None or float, optional
+        This parameter is passed to signal.clean. Please see the related
+        documentation for details
+
+    t_r: float, optional
+        This parameter is passed to signal.clean. Please see the related
+        documentation for details
+
+    memory: instance of joblib.Memory or string
+        Used to cache the masking process.
+        By default, no caching is done. If a string is given, it is the
+        path to the caching directory.
+
+    memory_level: integer, optional
+        Rough estimator of the amount of memory used by caching. Higher value
+        means more memory for caching.
+
+    n_jobs: integer, optional, default=1
+        The number of CPUs to use to do the computation. -1 means
+        'all CPUs', -2 'all CPUs but one', and so on.
+
+    verbose: integer, optional
+        Indicate the level of verbosity. By default, nothing is printed
+
+    References
+    ----------
+
+    """
+
+    def __init__(self, mask=None, n_components=20,
+                 smoothing_fwhm=6,
+                 standardize=True,
+                 random_state=0,
+                 target_affine=None, target_shape=None,
+                 low_pass=None, high_pass=None, t_r=None,
+                 alpha=1,
+                 n_iter=1000,
```

Comment:
Should the default be to 'auto'? If not, we should probably remove the option 'auto'.